### PR TITLE
Abstracting pod interface in kubernetes plugin to enable easier vendoring

### PR DESCRIPTION
**Pull request by vjsamuel**
When a user creates an Indexer outside of the beats repo, he/she needs to vendor in the k8s repo. This should ideally be unnecessary as the retrieval of Pods is abstracted from the user. This PR aims at adding a wrapper struct `Pod` which would be passed to Indexers to retrieve metadata.

### CODE REVIEW

File: libbeat/processors/kubernetes/podwatcher.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 
+	"encoding/json"
```
Comment (ruflin) [inline libbeat/processors/kubernetes/podwatcher.go]: some goimports fun ...

File: vendor/vendor.json [hunk 2]
```diff
@@ -160,187 +160,6 @@
 			"revisionTime": "2016-09-16T08:04:11Z"
 		},
 		{
-			"checksumSHA1": "K0iEPnt2DZL5/YrrzAQoMnA+9Pc=",
```
Comment (ruflin) [inline vendor/vendor.json]: 👍

**Comment by ruflin** [comment]
@exekias Could also please have a look?

**Comment by ruflin** [comment]
jenkins, test it

File: libbeat/processors/kubernetes/podwatcher.go (hunk 1 above)
Comment (vjsamuel) [inline libbeat/processors/kubernetes/podwatcher.go]: bleh! i hate it when it does that.

Comment (vjsamuel) [inline libbeat/processors/kubernetes/podwatcher.go]: updated.

File: filebeat/beater/filebeat.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/elastic/beats/filebeat/spooler"
 
 	// Add filebeat level processors
-	_ "github.com/elastic/beats/filebeat/processors/kubernetes"
+	_ "github.com/elastic/beats/filebeat/processor/kubernetes"
```
Comment (exekias) [inline filebeat/beater/filebeat.go]: I think we should keep consistency with libbeat (which uses `processors`)

File: libbeat/processors/kubernetes/podwatcher.go [hunk 4]
```diff
@@ -124,44 +125,64 @@ func (p *PodWatcher) Run() bool {
...
-	delete(p.annotationCache.pods, pod.Metadata.GetUid())
+	delete(p.annotationCache.pods, pod.Metadata.UID)
 
 	for _, index := range p.indexers.GetIndexes(pod) {
 		delete(p.annotationCache.annotations, index)
 	}
 }
 
+func (p *PodWatcher) getPodMeta(pod *corev1.Pod) *Pod {
+
```
Comment (exekias) [inline libbeat/processors/kubernetes/podwatcher.go]: Could you please remove the spurious empty lines at the beginning and end of this function?

File: filebeat/beater/filebeat.go (hunk 3 above)
Comment (ruflin) [inline filebeat/beater/filebeat.go]: @exekias I disagree on this one as we should change it for some time in libbeat to `processor`. It is `processors` for historical reasons.

Comment (exekias) [inline filebeat/beater/filebeat.go]: Ok then, it seems this was agreed some time ago :+1:

**Pull request merged**